app/src/main/java/io/github/dkter/aaaaa/AaaaaKeyboardView.kt

```diff
@@ -108,4 +109,21 @@ class AaaaaKeyboardView(
             this.keyboardListener.onReturn()
         }
     }
+
+    override fun onLongClick(v: View): Boolean {
+        val id = v.getId()
+        return if (id == R.id.btnA) {
+            this.keyboardListener.onLongA()
+            true
+        } else false
+    }
+
+    override fun onTouch(v: View?, event: MotionEvent?): Boolean {
+        val id = v?.id
+        val action = event?.action
+
+        if (id == R.id.btnA && action == MotionEvent.ACTION_UP) keyboardListener.onReleaseA()
```

Comment:
You don't have to implement this if you don't want, but I think it would be cool if for the keys that can be repeated, the first input action happens on the "down" touch event of the button instead of in the `onClick` event above. This would mirror more closely what happens on desktop keyboards (where the first key is input immediately, and the repetition happens after a delay) and would feel a little more responsive.

If you feel like that's outside the scope of this PR, that's fine, I can get around to this eventually.

Reply:
I'm not sure about this. I tried that out while working on this PR, but it led to some issues:

* the ripple on touch didn't appear anymore.
* the user might accidentally type two 'a's if they hold down for too long
* the code is a little bit more confusing
* we would have to override `performClick` on the 'a' button in order to allow visually impaired people to use this app (see [this](https://stackoverflow.com/questions/27462468/custom-view-overrides-ontouchevent-but-not-performclick#50343572); although I'm not sure if it really matters for a gimmick app)
* it doesn't resemble the behavior of other software keyboards

Most of these issues could be resolved, but I doubt that it's worth it.

Reply:
ah okay, thanks for trying it out